Plain `go test ./minio` can hang for 10 minutes: site replication cleanup runs before the TF_ACC skip

> **Update (post-#1031):** with no env vars set this no longer reproduces — the endpoint is empty and `minio.New("")` fails fast. The remaining case is the local-run setup the docs recommend: `MINIO_ENDPOINT` + credentials exported but `TF_ACC` unset. The cleanup still dials all three instances before the skip (and, if they are reachable with valid credentials, deletes their buckets). Updated "done when": with endpoints/credentials exported but `TF_ACC` unset, `go test ./minio` skips these tests without any network call; acceptance runs clean up as before. Original report below for context.

First of all, thanks for this provider and for the recent guidance on the open issues — much appreciated!

While working on #1020 I noticed that a plain `go test ./minio` (no acceptance env vars set, i.e. what someone would run for unit tests only) makes real network calls, and in my environment it hangs until the default `-timeout` kills the run after 10 minutes:

```
panic: test timed out after 10m0s
	running tests:
		TestAccMinioSiteReplication_update (1m24s)
```

The goroutine dump shows where it's stuck — bucket cleanup dialing a MinIO endpoint with empty credentials:

```
net/http.(*Client).Do(...)
github.com/minio/minio-go/v7.(*Client).ListBuckets(...)
github.com/aminueza/terraform-provider-minio/v3/minio.cleanupBucketsForEndpoint(0x..., {"minio:9000"}, {0x0, 0x0}, {0x0, 0x0})
github.com/aminueza/terraform-provider-minio/v3/minio.cleanupAllBuckets(0x...)
github.com/aminueza/terraform-provider-minio/v3/minio.TestAccMinioSiteReplication_update(0x...)
```

Root cause: every `TestAccMinioSiteReplication_*` test calls `cleanupAllBuckets(t)` as its first statement, *before* `resource.Test` gets a chance to skip on missing `TF_ACC` (and before `testAccPreCheckSiteReplication`'s own skip):

https://github.com/aminueza/terraform-provider-minio/blob/main/minio/resource_minio_site_replication_test.go#L94-L102

So even when every acceptance test is going to be skipped, the cleanup still tries to reach `minio:9000`, `secondminio:9000` and `thirdminio:9000`. In environments where those dials fail fast it "only" produces warning noise, but with minio-go's request retries against a slow/unreachable endpoint, the cleanups across the five site replication tests add up past the 10-minute default timeout and the whole run panics.

What I propose:

- Guard `cleanupAllBuckets` with the same condition `resource.Test` uses, e.g. `t.Skip` when `TF_ACC` is unset, so unit-only runs never touch the network.
- Optionally, bound each cleanup with a `context.WithTimeout` so a single unreachable instance can't stall the suite even in acceptance mode.

Done when: `go test ./minio` with no env vars set completes (with the acceptance tests skipped) without making network calls or hitting the test binary timeout.

I'm happy to send a PR for this if it sounds right.

--- a/minio/resource_minio_site_replication_test.go
+++ b/minio/resource_minio_site_replication_test.go
@@ -6,6 +6,7 @@ import (
 	"os"
 	"regexp"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
@@ -94,6 +95,13 @@ func testAccPreCheckSiteReplication(t *testing.T) {
 func cleanupAllBuckets(t *testing.T) {
 	t.Helper()
 
+	// resource.Test only skips on a missing TF_ACC after this cleanup has already
+	// run, so guard here too: without it, unit-only runs (TF_ACC unset) dial the
+	// configured endpoints and, with valid credentials, would wipe their buckets.
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("skipping site replication bucket cleanup: TF_ACC not set")
+	}
+
 	cleanupBucketsForEndpoint(t, testAccEndpoint(""), os.Getenv("MINIO_USER"), os.Getenv("MINIO_PASSWORD"))
 
 	cleanupBucketsForEndpoint(t, testAccEndpoint("SECOND_"), os.Getenv("SECOND_MINIO_USER"), os.Getenv("SECOND_MINIO_PASSWORD"))
@@ -111,7 +119,11 @@ func cleanupBucketsForEndpoint(t *testing.T, endpoint, accessKey, secretKey stri
 		return
 	}
 
-	ctx := context.Background()
+	// Bound each endpoint's cleanup so a slow or unreachable instance cannot
+	// stall the suite past the default 10m test binary timeout: minio-go retries
+	// with backoff, and this runs for 3 endpoints x 5 tests.
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
 	buckets, err := minioClient.ListBuckets(ctx)
 	if err != nil {
 		t.Logf("Warning: Failed to list buckets for %s: %v", endpoint, err)

